fastlane/lib/fastlane/helper/sh_helper.rb

```diff
@@ -25,20 +27,20 @@ def self.sh_control_output(command, print_command: true, print_command_output: t
       Encoding.default_external = Encoding::UTF_8
       Encoding.default_internal = Encoding::UTF_8
 
-      command = command.join(' ') if command.kind_of?(Array) # since it's an array of one element when running from the Fastfile
+      command = command.map(&:to_s).map(&:shellescape).join(' ') if command.kind_of?(Array) # since it's an array of one element when running from the Fastfile
       UI.command(command) if print_command
 
       result = ''
       if Helper.sh_enabled?
         exit_status = nil
-        IO.popen(command, err: [:child, :out]) do |io|
+        Open3.popen2e(command) do |stdin, io, thread|
```

Comment:
The reason for using `Open3` is to avoid the read-only global `$?`, which makes testing hard. Otherwise, there is no change here of any substance. With `IO.popen`, stdin and stdout (and I think stderr) are mapped to the single `io` in the block. With `popen2e`, `io` is the merged stdout and stderr. Standard input is mapped to a separate IO. In neither case is it used. The behavior of `sh` is unchanged with regard to input.